pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/VfExpressionTypeVisitor.java

```diff
@@ -4,39 +4,69 @@
 
 package net.sourceforge.pmd.lang.vf;
 
+import java.io.File;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.IdentityHashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.logging.Logger;
 
+import net.sourceforge.pmd.lang.ast.AbstractTokenManager;
 import net.sourceforge.pmd.lang.vf.ast.ASTAttribute;
 import net.sourceforge.pmd.lang.vf.ast.ASTAttributeValue;
+import net.sourceforge.pmd.lang.vf.ast.ASTCompilationUnit;
 import net.sourceforge.pmd.lang.vf.ast.ASTDotExpression;
 import net.sourceforge.pmd.lang.vf.ast.ASTElExpression;
 import net.sourceforge.pmd.lang.vf.ast.ASTElement;
 import net.sourceforge.pmd.lang.vf.ast.ASTExpression;
 import net.sourceforge.pmd.lang.vf.ast.ASTIdentifier;
 import net.sourceforge.pmd.lang.vf.ast.ASTText;
 import net.sourceforge.pmd.lang.vf.ast.VfParserVisitorAdapter;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+import net.sourceforge.pmd.properties.PropertyFactory;
+import net.sourceforge.pmd.properties.PropertySource;
 
 /**
  * Visits {@link ASTElExpression} nodes and stores type information for all {@link ASTIdentifier} child nodes.
  */
 public class VfExpressionTypeVisitor extends VfParserVisitorAdapter {
     private static final Logger LOGGER = Logger.getLogger(VfExpressionTypeVisitor.class.getName());
 
+    static final List<String> DEFAULT_APEX_DIRECTORIES = Collections.singletonList(".." + File.separator + "classes");
+    static final List<String> DEFAULT_OBJECT_DIRECTORIES = Collections.singletonList(".." + File.separator + "objects");
+
+    /**
+     * Directory that contains Apex classes that may be referenced from a Visualforce page.
+     */
+    public static final PropertyDescriptor<List<String>> APEX_DIRECTORIES_DESCRIPTOR =
```

Comment:
I was undecided if it was better to declare this here or in `VfParserOptions`. I chose here because it is closer to its usage.

Where should I add documentation for these new properties and the fact that they can be set via environment variables?